presto-main/src/main/java/io/prestosql/sql/planner/plan/JoinNode.java

```diff
@@ -136,12 +135,6 @@ public JoinNode(
                     "%s join do not work with %s distribution type",
                     type,
                     distributionType.get());
-            // It does not make sense to PARTITION when there is nothing to partition on
-            checkArgument(
```

Comment:
move the check to `io.prestosql.sql.planner.optimizations.AddExchanges.Rewriter#planPartitionedJoin` instead.